### PR TITLE
Revert "Fix slow execution when many breakpoints are used (#14953)" 

### LINKED ISSUE #4668: Can't debug when using PowerShell 7.4
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all [_open and closed_ issues](https://github.com/PowerShell/vscode-powershell/issues?q=is%3Aissue) to ensure it has not already been reported.
- [X] I have read the [troubleshooting](https://github.com/PowerShell/vscode-powershell/blob/main/docs/troubleshooting.md) guide.
- [X] I am sure this issue is with the _extension itself_ and does not reproduce in a standalone [PowerShell](https://github.com/PowerShell/PowerShell/issues/new/choose) instance.
- [X] I have verified that I am using the latest version of Visual Studio Code and the PowerShell extension.
- [X] If this is a security issue, I have read the [security issue reporting guidance](https://github.com/PowerShell/vscode-powershell/blob/main/SECURITY.md).

### Summary

It seems that VSCode doesn't stop on breakpoints when debugging in PS 7.4

### PowerShell Version

```console
Name                           Value
----                           -----
PSVersion                      7.4.0-preview.4
PSEdition                      Core
GitCommitId                    7.4.0-preview.4
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0.}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visual Studio Code Version

```console
1.80.1
x64
```

### Extension Version

```console
ms-vscode.powershell@2023.6.0
```

### Steps to Reproduce

1. Create empty file
2. Add some code

```

Write-Host "Test 1"

Write-Host "Test 2"
```

3. Set breakpoint

4. Press F5
5. Notice it doesn't stop

### Visuals

Debugging not working

### Logs

_No response_

### PR DESCRIPTION
**Pull request by andyleejordan**
# PR Summary

This reverts commit https://github.com/PowerShell/PowerShell/commit/d8decdc1944444efa811d465747c6e9b91b86b82.

This commit broke the VS Code extension's debugger, and should be
reverted until such time that the root cause is found and a fix applied.

Also disable false assertion that blocks pwsh on Parallels VM.

## PR Context

The resolves https://github.com/PowerShell/vscode-powershell/issues/4668, where the VS Code extension's debugger no longer worked with preview.4. First I confirmed that it worked with preview.3, then I cloned and built from source. After looking at the diff between preview.3 and preview.4 for changes that affected the debugger in the engine, I only found https://github.com/PowerShell/PowerShell/pull/14953. While I've looked at the changes in that PR, I haven't yet isolated _what_ broke things; but I reverted the merge commit and confirmed the debugger works again.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [x] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [x] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [x] Issue filed: Resolves https://github.com/PowerShell/vscode-powershell/issues/4668
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/System.Management.Automation/engine/Interop/Windows/QueryDosDevice.cs [hunk 1]
```diff
@@ -88,7 +88,8 @@ internal static string GetDosDeviceForNetworkPath(char deviceName)
                         }
                         else
                         {
-                            Diagnostics.Assert(false, "Really it is a dead code since GetDosDevice() is called only if PSDrive.DriveType == DriveType.Network");
+                            // TODO: This actually hit in a Parallels VM, so it's not quite dead yet!
+                            // Diagnostics.Assert(false, "Really it is a dead code since GetDosDevice() is called only if PSDrive.DriveType == DriveType.Network");
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/Interop/Windows/QueryDosDevice.cs:92]: Please open an issue about this and mark it "7.4-Blocking".

Comment (andyleejordan) [reply src/System.Management.Automation/engine/Interop/Windows/QueryDosDevice.cs:92]: Which commit, the debugger being broken or the executable failing on my VM, or both?

**Comment by andyleejordan** [comment]
@daxian-dbw filed https://github.com/PowerShell/PowerShell/issues/20043 and https://github.com/PowerShell/PowerShell/issues/20044

File: src/System.Management.Automation/engine/Interop/Windows/QueryDosDevice.cs (hunk 1 above)
Comment (iSazonov) [inline src/System.Management.Automation/engine/Interop/Windows/QueryDosDevice.cs:92]: This is a fallback from another p/invoke GetUNCForNetworkDrive - it is main code path. This says that something can work incorrectly on Parallels VM.
So I suggest investigate the issue in separate PR.

Comment (daxian-dbw) [reply src/System.Management.Automation/engine/Interop/Windows/QueryDosDevice.cs:92]: Open an issue about the fact that this assertion is wrong on Parallels VM. This code path needs to be reviewed as the assumption is incorrect. Can you please find out what the `PSDrive.DriveType` is when running in Parallel VM?

**Comment by fflaten** [comment]
Close this? See #20046 for breakpoints-fix. Parallels issue tracked in #20043

**Comment by daxian-dbw** [comment]
Superseded by #20046

File: src/System.Management.Automation/engine/Interop/Windows/QueryDosDevice.cs (hunk 1 above)
Comment (daxian-dbw) [reply src/System.Management.Automation/engine/Interop/Windows/QueryDosDevice.cs:92]: @andyleejordan Can you please check what `PSDrive.DriveType` is when the assertion fails in Parallel VM and share that info in #20043? Thanks!

**Pull request closed without merging**